test/integration/test_download.py

```diff
@@ -95,6 +95,15 @@ def testDownloadNonExistingArtifact(self):
         with pytest.raises(AttributeError):
             self.download('-n', '--release', '5.3', '--ice', '3.3')
 
+    def testDownloadBuildNumber(self):
+        # Old Jenkins artifacts are deleted so we can't download.
+        # Instead assert that an AttributeError is raised.
+        # This is not ideal since this error could occur for other reasons.
+        branch = self.branch + ':600'
```

Comment:
To test the build number logic, could we make use of the Jenkins API and retrieve the last successful build number using https://ci.openmicroscopy.org/job/OMERO-DEV-latest/lastSuccessfulBuild/buildNumber